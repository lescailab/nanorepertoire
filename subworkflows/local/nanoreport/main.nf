// This subworkflows is used to create the report of nanorepertoire analysis
// modules to include in this subworkflow

include { REPORT    }             from '../../../modules/local/report/main.nf'

workflow NANOREPORT {

    // with take we define the input channels

    take:
    report
    loop_tree
    clustersummaries
    cdr3histograms
    cdr3tables
    metadata



    main:

    REPORT(report, loop_tree, clustersummaries, cdr3histograms, cdr3tables, metadata)



    emit:
    analysis_report    = REPORT.out.report              // channel: [ val(meta), [ bai ] ]
    //versions           = REPORT.out.versions          // channel: [ versions.yml ]
}

