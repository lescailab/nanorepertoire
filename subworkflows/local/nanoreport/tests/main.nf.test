// nf-core subworkflows test nanoreport
nextflow_workflow {

    name "Test Subworkflow NANOREPORT"
    script "../main.nf"
    workflow "NANOREPORT"

    tag "subworkflows"
    tag "subworkflows_nfcore"
    tag "subworkflows/nanoreport"
    tag "samtools"
    tag "samtools/sort"
    tag "samtools/index"


    test("sarscov2 - bam - single_end") {

        when {
            workflow {
                """
                input[0] =[
                        file("https://raw.githubusercontent.com/DavideBag/testdata/dev/nanobodies/report/nftest_1_fastqc.zip", checkIfExists: true)
                        ]
                input[1] = [
                        file("https://raw.githubusercontent.com/DavideBag/testdata/dev/nanobodies/report/nftest.cutadapt.log", checkIfExists: true)
                        ]
                input[2] = [
                        file("$projectDir/assets/analysis_report.qmd")
                        ]
                input[3] = [
                        file("$projectDir/assets/loop_tree.qmd")
                        ]
                input[4] = [
                        file('https://raw.githubusercontent.com/DavideBag/testdata/dev/nanobodies/report/test_clusters.summary', checkIfExists: true)
                        ]
                input[5] = [
                        file('https://raw.githubusercontent.com/DavideBag/testdata/dev/nanobodies/report/test_cdr3.hist', checkIfExists: true)
                        ]
                input[6] = [
                        file('https://raw.githubusercontent.com/DavideBag/testdata/dev/nanobodies/report/test_cdr3.tsv', checkIfExists: true)
                        ]
                input[7] = [
                    [id: 'test', sampleID: 'sample1', individualID: 'ind1', immunisation: 'imm1', boost: 'boost1']
                        ]
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(workflow.out).match()}
            )
        }
    }
}
