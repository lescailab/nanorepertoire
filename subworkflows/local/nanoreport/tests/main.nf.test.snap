{
    "sarscov2 - bam - single_end": {
        "content": [
            {
                "0": [
                    "multiqc_report.html:md5,976c73b904ace61615a0fb7ae6ccc500"
                ],
                "1": [
                    "multiqc_report.html:md5,394254320bfd432190c74be4db316625"
                ],
                "2": [
                    "analysis_report.html:md5,3f48f4bf0ed94e8ed7f4ba4c6fe38350"
                ],
                "3": [
                    "versions.yml:md5,6bca34a522f2030c562c7a521ce8264b",
                    "versions.yml:md5,9f153f756e74bd1c51ff66be000f5ae4"
                ],
                "analysis_report": [
                    "analysis_report.html:md5,3f48f4bf0ed94e8ed7f4ba4c6fe38350"
                ],
                "multiqc_report": [
                    "multiqc_report.html:md5,976c73b904ace61615a0fb7ae6ccc500"
                ],
                "multiqc_report1": [
                    "multiqc_report.html:md5,394254320bfd432190c74be4db316625"
                ],
                "versions": [
                    "versions.yml:md5,6bca34a522f2030c562c7a521ce8264b",
                    "versions.yml:md5,9f153f756e74bd1c51ff66be000f5ae4"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-18T10:44:17.744463"
    }
}